BarrysPeanutsTemporalJava/src/main/java/barryspeanuts/App.java

```diff
@@ -0,0 +1,127 @@
+    // implementations.
+    Worker worker = factory.newWorker(TASK_QUEUE);
+
+    // Workflows are stateful. So you need a type to create instances.
+    worker.registerWorkflowImplementationTypes(ShoppingCartWorkflowImpl.class);
+
+    // Start all workers created by this factory.
+    factory.start();
+    logger.info("Worker started for task queue: {} with WorkflowID : {}.", TASK_QUEUE, WORKFLOW_ID);
+
+    // now we can start running instances of our workflow - its state will be persisted
+    WorkflowOptions options =
+        WorkflowOptions.newBuilder()
+            .setTaskQueue(TASK_QUEUE)
+            .setWorkflowId(WORKFLOW_ID)
+            // set the return options
+            .setRetryOptions(
+                RetryOptions.newBuilder()
+                    .setInitialInterval(Duration.ofSeconds(1))
+                    .setMaximumInterval(Duration.ofSeconds(10))
+                    .build())
+            .build();
+
+    ShoppingCartWorkflow wf = client.newWorkflowStub(ShoppingCartWorkflow.class, options);
+    try {
+      Address address =
+          new Address("123 Main Street", "Apt 1", "Anytown", "CA", "99999-9999", "USA");
+
+      Customer customer =
+          new Customer(
+              UUID.randomUUID(),
+              "Josiah",
+              "Bartlet",
+              "prez@whitehouse.gove",
+              "202 456 1414",
+              address);
+
+      WorkflowClient.start(wf::startWorkflow);
+
+      // Create purchase items and add them to the Shopping Cart
+      List<PurchaseItem> purchaseItems = new ArrayList<>();
+
+      purchaseItems.add(
+          new PurchaseItem(
+              UUID.randomUUID(),
+              customer,
+              "Barry's Deluxe Peanuts:",
+              3,
+              new BigDecimal("12.99"),
+              new BigDecimal("5")));
+
+      purchaseItems.add(
+          new PurchaseItem(
+              UUID.randomUUID(),
+              customer,
+              "Barry's Smoked Peanuts",
+              5,
+              new BigDecimal("25.99"),
+              new BigDecimal("2")));
+
+      purchaseItems.add(
+          new PurchaseItem(
+              UUID.randomUUID(),
+              customer,
+              "Barry's Soft Shell Peanuts",
+              1,
+              new BigDecimal("9.99"),
+              new BigDecimal("1")));
+
+      wf.addItems(purchaseItems);
+
+      // Get the customer information for the credit card
+      String firstName = customer.getFirstName();
+      String lastName = customer.getLastName();
+      CreditCard creditCard =
+          new CreditCard(firstName + " " + lastName, "1111222233334444", 8, 26, 111);
+
+      // Checkout using the customer's base address as the billing and shipping addresses
+      CheckoutInfo checkoutInfo =
+          new CheckoutInfo(creditCard, customer.getAddress(), customer.getAddress(), "FEDEX");
+
+      wf.checkOut(checkoutInfo);
+
+      // Exit the workflow
+      wf.exit();
```

Comment:
Nice! Very simple. Arguably the workflow could complete on checkout, but I suppose making a separate exit is fine too. Can do the same for the actor side also...

Reply:
+1 for making the workflow exit automatically.